activestorage/lib/active_storage/validations/size_validator.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  module Validations
+    class SizeValidator < ActiveModel::EachValidator
+      delegate :number_to_human_size, to: ActiveSupport::NumberHelper
+
+      AVAILABLE_CHECKS = [:less_than, :less_than_or_equal_to, :greater_than, :greater_than_or_equal_to, :between]
+
+      def check_validity!
+        unless (AVAILABLE_CHECKS).any? { |argument| options.has_key?(argument) }
+          raise ArgumentError, "You must pass either :less_than, :greater_than, or :between to the validator"
+        end
+      end
+
+      def validate_each(record, attribute, value)
+        files = record.send(attribute)
+        # only attached
+        return true unless files.attached?
+
+        files = Array.wrap(files)
+
+        files.each do |file|
+          if content_size_valid?(file)
+            record.errors.add(attribute, options[:message].presence || "size #{number_to_human_size(file.blob.byte_size)} is not between required range")
+            return
+          end
+        end
+      end
+
+      def content_size_valid?(file)
```

Comment:
It's not immediately obvious to me what `size: { greater_than: 100.kilobytes }` does, I assume it means the minimum file size is 100kb?

I find the DSL here kind of confusing, compare the ActiveRecord validations on length with these ActiveStorage file size validations:

```ruby
# ActiveRecord
validates :name, length: { minimum: 4, maximum: 20 }
validates :name, length: { minimum: 2 }
validates :bio, length: { maximum: 500 }
validates :password, length: { in: 6..20 }
validates :registration_number, length: { is: 6 }

# ActiveStorage
validates :preview, attached: true, size: { greater_than: 100.kilobytes }
validates :preview, attached: true, size: { between: 0..100.kilobytes }
validates :preview, attached: true, size: { less_than: 100.megabytes }
```

Shouldn't the ActiveStorage validators use `maximum`, `minimum`, and `in` like other validations? (I don't think `is` would be very useful for file size).

Reply:
I guess I don’t really see the usefulness of >= or <=. As for between, wouldn’t that be covered by `in`?

Reply:
I agree. When comparing numericality it makes sense, in mathematics we need these functions, however, when validating a file size it's a little overkill. IRL I would never specify `this file more than 100kb` (101kb or more). It's always going to be `should be at least 100kb no more than 1mb`.

Reply:
I think it's actually not useful to have less/greater than. maximum/minimum are replacements for less/greater_than_or_equal_to

e.g.
`file max 100kb` is the same as `file <= 100kb` not `file < 100kb`